js/ui/marker.js

```diff
@@ -13,6 +13,10 @@ var Point = require('point-geometry');
  * @param {HTMLElement=} element DOM element to use as a marker (creates a div element by default)
  * @param {Object=} options
  * @param {PointLike=} options.offset The offset in pixels as a [`PointLike`](#PointLike) object to apply relative to the element's top left corner. Negatives indicate left and up.
+ * @param {boolean=} [options.pitchScale=false] Whether the marker should scale smaller into the background or larger into the foreground when the map is pitched and rotated
+ * @param {boolean=} [options.zoomScale=false] Whether the marker should scale with the map as you zoom in and out
+ * @param {number=} [options.nativeZoom=map.getZoom()] The native zoom level of the marker. If the map is at this zoom the marker will not be scaled. If zoom is greater the marker will be enlarged. If zoom is less the marker will be shrunk.
+ * @param {number=} [options.zoomFactor=1.6] The multiple by which to scale the size of the marker between each zoom level
```

Comment:
> `stops: [[1, 1/512], [20, 1024]]`

 > I don't think that works unless you want your native scale to be 10

With a little math, you can adjust your "native scale" to be whatever you choose. 

> I don't really see a good way to add this without completely hacking apart `mapbox-gl-function`

Let's hack it! 😄 This is a highly requested feature for `mapbox-gl-function`. I'm more than happy to help you with that PR.

Some syntax ideas: 

```
// use a value which represents 500m of "real world" distance (would require more work)
{
    type: 'unit',
    value: 500
    units: 'meters' 
}

// at zoom 5, use a value of 1. scale the value proportionally with the zoom
{
    type: 'zoom-interpolated',
    zoom: 5,
    value: 1       
}
```